src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/SortClauseBuilder.java

```diff
@@ -64,178 +57,58 @@ public SortClause build(JsonNode node) {
-        if (totalFields > 1) {
-          throw ErrorCodeV1.VECTOR_SEARCH_USAGE_ERROR.toApiException();
-        }
-        if (!inner.getValue().isTextual()) {
-          throw ErrorCodeV1.SHRED_BAD_VECTORIZE_VALUE.toApiException();
-        }
+      sortExpressions.add(
+          buildAndValidateExpression(validateSortClausePath(path), inner.getValue(), totalFields));
+    }
+    return new SortClause(sortExpressions);
+  }
 
-        String vectorizeData = inner.getValue().textValue();
-        if (vectorizeData.isBlank()) {
-          throw ErrorCodeV1.SHRED_BAD_VECTORIZE_VALUE.toApiException();
-        }
-        SortExpression exp = SortExpression.vectorizeSearch(vectorizeData);
-        sortExpressions.clear();
-        sortExpressions.add(exp);
-        // TODO: aaron 17-oct-2024 - this break seems unneeded as above it checks if there is only 1
-        // field, leaving for now
-        break;
-      } else if (inner.getValue().isArray()) {
-        // TODO: HACK: quick support for tables, if the value is an array we will assume the column
-        // is a vector then need to check on table pathway that the sort is correct.
-        // NOTE: does not check if there are more than one sort expression, the
-        // TableCqlSortClauseResolver will take care of that so we can get proper ApiExceptions
-        // this is also why we do not break the look here
-        sortExpressions.add(
-            SortExpression.tableVectorSort(path, arrayNodeToVector((ArrayNode) inner.getValue())));
-      } else if (inner.getValue().isTextual()) {
-        // TODO: HACK: quick support for tables, if the value is an text  we will assume the column
-        // is a vector and the user wants to do vectorize then need to check on table pathway that
-        // the sort is correct.
-        // NOTE: does not check if there are more than one sort expression, the
-        // TableSortClauseResolver will take care of that so we can get proper ApiExceptions
-        // this is also why we do not break the look here
-        sortExpressions.add(SortExpression.tableVectorizeSort(path, inner.getValue().textValue()));
-      } else {
-        if (!inner.getValue().isInt()
-            || !(inner.getValue().intValue() == 1 || inner.getValue().intValue() == -1)) {
-          throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
-              "Sort ordering value can only be `1` for ascending or `-1` for descending (not `%s`)",
-              inner.getValue());
-        }
+  protected abstract SortExpression buildAndValidateExpression(
+      String validatedPath, JsonNode innerValue, int totalFields);
 
-        boolean ascending = inner.getValue().intValue() == 1;
-        SortExpression exp = SortExpression.sort(validatedPath, ascending);
-        sortExpressions.add(exp);
-      }
+  protected SortExpression defaultBuildAndValidateExpression(
+      String validatedPath, JsonNode innerValue) {
```

Comment:
have not read the whole PR yet - but having a default* in the base feels odd. is there a reason we don't use the same name and have the sub call the super if it wants to ?

Reply:
reading further, this looks like handling for a regular sort expression so maybe rename ?

Reply:
Yeah it's bit awkward; results of incremental conversion.
Let me think about how to change.

Reply:
and yes, I should rename "validatedPath" as just "path"; no need to document in name (if need be, in Javadoc param definition)